src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,85 @@
+import '../../App.scss';
+import React, { useEffect, useState } from 'react';
+import { Loader } from '../Loader';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { PersonBody } from '../PersonBody/PersonBody';
+import { useParams } from 'react-router-dom';
+import { getPreparedPeople } from '../../Utils/GetPreparedPeople';
+
+interface Props {}
+
+export const PeoplePage: React.FC<Props> = ({}) => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loadFail, setLoadFail] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const { chosenId } = useParams();
+
+  const isCompletedLoadingWithoutFails =
+    !isLoading && !people.length && !loadFail;
+
+  const isSuccessfullyLoadedListWithPeople = !isLoading && !!people.length;
+
+  useEffect(() => {
+    setIsLoading(true);
+    setLoadFail(false);
+    getPeople()
+      .then(res => {
+        setPeople(getPreparedPeople(res));
+      })
+      .catch(() => setLoadFail(true))
+      .finally(() => {
+        setIsLoading(false);
+      });
```

Comment:
It's better to move the logic inside the `useEffect `hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep `useEffect `focused on side effects.